presto-main/src/main/java/com/facebook/presto/spiller/FileSingleStreamSpillerFactory.java

```diff
@@ -93,35 +93,22 @@ public FileSingleStreamSpillerFactory(
             boolean spillCompressionEnabled,
             boolean spillEncryptionEnabled)
     {
-        this.spillStorageService = requireNonNull(spillStorageService, "spillStorageService is null");
+        this.storageServiceManager = requireNonNull(storageServiceManager, "storageServiceManager is null");
         this.serdeFactory = new PagesSerdeFactory(requireNonNull(blockEncodingSerde, "blockEncodingSerde is null"), spillCompressionEnabled);
         this.executor = requireNonNull(executor, "executor is null");
         this.spillerStats = requireNonNull(spillerStats, "spillerStats can not be null");
         requireNonNull(spillPaths, "spillPaths is null");
         this.spillPaths = ImmutableList.copyOf(spillPaths);
 
-        spillPaths.forEach(path -> {
-            try {
-                spillStorageService.createDirectories(path);
-            }
-            catch (IOException e) {
-                throw new IllegalArgumentException(
-                        format("could not create spill path %s; adjust experimental.spiller-spill-path config property or filesystem permissions", path), e);
-            }
-            if (!spillStorageService.canWrite(path)) {
-                throw new IllegalArgumentException(
-                        format("spill path %s is not writable; adjust experimental.spiller-spill-path config property or filesystem permissions", path));
-            }
-        });
         this.maxUsedSpaceThreshold = maxUsedSpaceThreshold;
         this.spillEncryptionEnabled = spillEncryptionEnabled;
         this.roundRobinIndex = 0;
     }
 
-    @PostConstruct
+    @VisibleForTesting
```

Comment:
It's not very clear to me why some initialization is putting in constructor, some are putting here with `@ PostConstruct`. For now i move all of them into `initialize()`

Reply:
Yeah, it makes sense. It feels like the cleanup should be happening on server start, not on server shutdown. Assuming there's no bugs we shouldn't leak files in case of a normal server termination.

Reply:
@arhimondr : It's still possible, say the server power get suddenly power off right?